django_sorcery/db/query.py

```diff
@@ -33,6 +70,61 @@ def get(self, *args, **kwargs):
 
         return super(Query, self).get(*args, **kwargs)
 
+    def filter(self, *args, **kwargs):
+        """
+        Standard SQLAlchemy filtering plus django-like expressions can be provided:
+
+        For example::
+
+            MyModel.objects.filter(MyModel.id == 5)
+            MyModel.objects.filter(id=5)
+            MyModel.objects.filter(id__gte=5)
+            MyModel.objects.filter(relation__id__gte=5)
+        """
+        args = args + tuple(self._lookup_to_expression(k, v) for k, v in kwargs.items())
+        return super(Query, self).filter(*args)
+
+    def _lookup_to_expression(self, lookup, value):
+        parts = lookup.split(LOOKUP_SEP)
+        info = meta.model_info(self._only_full_mapper_zero("get"))
+
+        props = dict(info.column_properties)
+        lhs = None
+
+        for i, part in enumerate(parts, 1):
+            is_last = i == len(parts)
+
+            if part in LOOKUP_TO_EXPRESSION:
+                return LOOKUP_TO_EXPRESSION[part](lhs, value)
+
+            elif part in info.relationships:
+                rel = info.relationships[part]
+                pairs = rel.local_remote_pairs_for_identity_key
+
+                # directly comparing to model
+                # e.g. .filter(relation=instance)
+                # which means we should filter by FK to avoid additional joins
+                if is_last and isinstance(value, ModelBase) and pairs:
+                    return sa.and_(*[local == getattr(value, remote.key) for local, remote in pairs])
+
+                lhs = rel.related_model
+                props = dict(meta.model_info(lhs).column_properties)
+
+            elif part in info.composites:
+                comp = info.composites[part]
+                props = comp.properties
+
+                # directly comparing to composite
+                # e.g. .filter(composite=instance)
```

Comment:
never seen a composite being used like that though it is possible to filter by one of composite properties.